src/engine/enginemaster.cpp

```diff
@@ -502,27 +662,28 @@ void EngineMaster::process(const int iBufferSize) {
         // Submit master samples to the side chain to do broadcasting, recording,
         // etc. (cpu intensive non-realtime tasks)
         if (m_pEngineSideChain != NULL) {
-            if (m_pMasterTalkoverMix->toBool()) {
-                // Add Master and Talkover to Sidechain output, re-use the
-                // talkover buffer
-                // Note: m_ppSidechain = &m_pTalkover;
-                SampleUtil::addWithGain(m_pTalkover,
-                        m_pMaster, 1.0,
-                        iBufferSize);
+            if (configuredTalkoverMixMode == TalkoverMixMode::DIRECT_MONITOR
+                && m_pNumMicsConfigured->get() > 0) {
+                // When using direct monitoring, the user hears the microphone inputs
+                // mixed with the master output in hardware without the latency of
+                // sending the input signal into Mixxx. However, to record & broadcast
+                // the input signals mixed with the master mix, they need to be sent
+                // into Mixxx. The signal Mixxx receives is delayed by the input latency,
+                // so to keep the input and master mix aligned, the recording/broadcasting
+                // signal needs to be delayed, but the master/booth outputs must not be
+                // delayed.
+                SampleUtil::copy(m_pSidechain, m_pMaster, iBufferSize);
+                m_pInputLatencyCompensationDelay->process(m_pSidechain, iBufferSize);
```

Comment:
The talkover mix should be mixed into m_pSidechain here. I suspect this was another confusion switching betwen Git branches while I was developing this.